Sources/MapboxNavigation/String.swift

```diff
@@ -39,4 +39,20 @@ extension String {
         }
         return digest.lazy.map { String(format: "%02x", $0) }.joined()
     }
+    
+    /**
+     Calculate the Jaccard Similarity of current string with a string set and count.
+     */
+    public func calculateJaccardSimilarity(compareSet: Set<String>, allCounts: Int, separators: CharacterSet = CharacterSet(charactersIn: " ")) -> Double {
```

Comment:
Jaccard similarity seems to be appropriate except that the Mapbox Streets source abbreviates words in English, which can frustrate any word-by-word comparison. We’re using Levenshtein–Damerau edit distance algorithm for finding matching route summary strings; it goes character by character:

https://github.com/mapbox/mapbox-navigation-ios/blob/69370c2788f2e03d88e43699a060d36a415efaa5/Sources/MapboxCoreNavigation/MinimumEditDistance.swift#L5

Both algorithms will disambiguate the most blatant differences:

Electronic horizon | Feature querying | Jacard | Levenshtein–Damerau
----|----|----|----
Main Street | Main St | **0.50** | **4**
Main Street | W Elm St | 0 | 9

Electronic horizon | Feature querying | Jacard | Levenshtein–Damerau
----|----|----|----
14th Street Northwest | 14th St NW | **0.33** | **12**
14th Street Northwest | 8th St NW | 0 | 14

But Jacard similarity won’t detect differences in abbreviated words:

Electronic horizon | Feature querying | Jacard | Levenshtein–Damerau
----|----|----|----
[West Peachtree Street Northwest](https://www.openstreetmap.org/#map=19/33.78079/-84.38759) | W Peachtree St NW | **0.25** | **15**
West Peachtree Street Northwest | Peachtree Pl NE | **0.25** | 19

Electronic horizon | Feature querying | Jacard | Levenshtein–Damerau
----|----|----|----
Franklin Street | Franklin St | **0.5** | **4**
Franklin Street | Franklin Ct | **0.5** | 5

The basic Levenshtein algorithm would be more appropriate than Levenshtein–Damerau, since the latter considers transpositions whereas these street names would only ever be abbreviated.

Reply:
Thanks for the suggestion, I'll use the `func minimumEditDistance(to word: String) -> Int` for the similarity comparison. And it's also concerning that the names from the Electronic horizon road graph would be an Array of road name strings, instead of a single one. How do we handle this issue?

Reply:
Typically, the road object will only have multiple names because the subsequent names are actually route numbers. It’s possible for a road to have multiple names per se, but that’s relatively very rare. It would be totally reasonable to compare only the road object’s first name (not route number).

Reply:
Thanks for the suggestion, I'll use the first name to compare.

Reply:
After the tests, the following cases require all the strings inside the `electronicHorizonRoadNames` to be considered:
```
electronicHorizonRoadNames : ["I 280", "Junipero Serra Freeway"]
Feature querying: ["El Monte Rd", "Junipero Serra Fwy"]
Only consider the first one: final label: "El Monte Rd"
```